twine/__main__.py

```diff
@@ -25,8 +26,24 @@
 def main() -> Any:
     try:
         return cli.dispatch(sys.argv[1:])
-    except (exceptions.TwineException, requests.HTTPError) as exc:
-        return _format_error(f"{exc.__class__.__name__}: {exc.args[0]}")
+    except requests.HTTPError as exc:
+        return _format_error(_format_http_exception(exc))
+    except exceptions.TwineException as exc:
+        return _format_error(_format_exception(exc))
+
+
+def _format_http_exception(exc: requests.HTTPError) -> str:
+    status_code = exc.response.status_code
+    status_phrase = http.HTTPStatus(status_code).phrase
+    return (
+        f"{exc.__class__.__name__} from {exc.response.url}: "
+        f"{status_code} {status_phrase}\n"
+        f"{exc.response.reason}"
+    )
```

Comment:
Using [`HTTPStatus`](https://docs.python.org/3/library/http.html#http.HTTPStatus) from the standard library.

Codes and phrases: https://github.com/python/cpython/blob/935586845815f5b4c7814794413f6a812d4bd45f/Lib/http/__init__.py#L69